utilities/verify-db-migration.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+# Stop exiting and return non-zero error code if an error occurs
+set -ue
+
+SCRIPTPATH="$(
+  cd -- "$(dirname "$0")" >/dev/null 2>&1 || exit
+  pwd -P
+)"
+
+REPO_ROOT="$SCRIPTPATH/.."
+
+# Checks if a binary is present on the local system
+exit_if_binary_not_installed() {
+    for binary in "$@"; do
+        command -v "$binary" >/dev/null 2>&1 || {
+            echo >&2 "Script requires '$binary' command-line utility to be installed on your local machine. Aborting..."
+            exit 1
+        }
+    done
+}
+
+cd "$REPO_ROOT/utilities/db-migration"
+go mod download
+echo "Start PostgreSQL"
+
+cd "$REPO_ROOT"
+USE_MASTER_SCHEMA="false" ./create-dev-env.sh
+make db-drop
+make db-migrate
+
+MASTER_SQL_FILE="$(mktemp)"
+echo "MASTER_SQL_FILE: $MASTER_SQL_FILE"
+
+MIGRATION_SQL_FILE="$(mktemp)"
+echo "MIGRATION_SQL_FILE: $MIGRATION_SQL_FILE"
+
+exit_if_binary_not_installed "pg_dump"
+
+echo "Create a pg_dump file for master schema"
+pg_dump postgresql://postgres:gitops@localhost:5432/postgres > "$MASTER_SQL_FILE" && cat "$MASTER_SQL_FILE"
+
+echo "Check if downgrading and upgrading by a single version leads to same schema."
+make db-migrate-downgrade
+make db-migrate-upgrade
+pg_dump postgresql://postgres:gitops@localhost:5432/postgres > "$MIGRATION_SQL_FILE" && cat "$MIGRATION_SQL_FILE"
+diff -Nur "$MIGRATION_SQL_FILE" "$MASTER_SQL_FILE"
+
+echo "Check if downgrading and upgrading by a two versions leads to same schema."
+make db-migrate-downgrade
+make db-migrate-downgrade
+make db-migrate-upgrade
+make db-migrate-upgrade
```

Comment:
LGTM as well, thanks for catching the typo, Panos!
The only comment from my side, is this a redundant check and needed(?)

My quick suggestion would be to either keep one of them.

Reply:
I had discussed this with Jonathan before going on my vacation. Just to be on the safer side, we're checking if last 2 most recent migrations work properly after doing initial single migration check.